src/utils/RuntimeEnvironment.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * This is a subset of the full DOM Location interface,
+ * containing only the properties we need, for easier testing.
+ */
+export interface LocationSlice {
+	search: string;
+}
+
+export interface RuntimeEnvironment {
+	isInDevMode: boolean;
+	runsInDevEnvironment: boolean;
+}
+
+export class UrlRuntimeEnvironment implements RuntimeEnvironment {
+	private _isInDevMode: boolean;
+	private _runsInDevEnvironment: boolean;
```

Comment:
what are the different purposes of these two fields?

Reply:
One marks the `?devMode` query for previewing the banners on WPORG, the other is the `?devBanner` query item from the Fun Forge